src/client/headers/shared/types.hpp

```diff
@@ -1354,6 +1258,179 @@ namespace intercept {
+            };
+
+            // "code" abstraction, useful for placing on callstack as an item, once again, Usual Intercept users won't need this and shouldn't use this
+            class callstack_item_data : public callstack_item {
+            private:
+                void* operator new(std::size_t sz_);
+                void operator delete(void* ptr_, std::size_t sz_);
+            public:
+                ref<game_data_code> _code;
+                //actual var space of the scope itself
+                game_var_space _dataVarSpace;
+                int _programCounter;
+                callstack_item_data(){}
+                callstack_item_data(game_data_code* code, callstack_item* parent, game_var_space varSpace, int stackPos, const game_state* gs);
+
+                virtual game_instruction* next(int& d1, const game_state* s) {
+                    const auto_array<ref<game_instruction>> &instructions = _code.get()->instructions;
+                    if (_programCounter >= instructions.size()) {
+                        d1 = execution::done;
+                        return nullptr;
+                    } else {
+                        d1 = execution::continueEx;
+                        return instructions[_programCounter++];
+                    }
+                }
+
+                virtual bool someEH(void* state) {
+                    return false;
+                }
+                virtual bool someEH2(void* state) {
+                    return false;
+                }
+
+                virtual r_string get_type() const {
+                    return r_string();
+                }
+
+                virtual serialization_return serialize(param_archive& ar) {
+                    return serialization_return::no_error;
+                }
+
+                virtual void on_before_exec() {}
+
+                virtual const char* getName() const {
+                    return "Frame";
+                };
+                virtual int varCount() const {
+                    return 1;
+                }
+                virtual int getVariables(const IDebugVariable** storage, int count) const {
+                    return 1;
+                }
+                /**
+                 * @brief EvaluateExpression doesn't work, it will throw an exception if called
+                */
+                virtual __internal::I_debug_value::RefType EvaluateExpression(const char* code, unsigned int rad) {
+                    throw std::bad_function_call();
+                    return game_value(1).data.get();
+                }
+                virtual void getSourceDocPosition(char* file, int fileSize, int& line) {
+                    file[0] = 0;
+                    line = 0;
+                }
+                virtual IDebugScope* getParent() {
+                    return _parent;
+                }
+            };
+
+            auto add_callstack_item(ref<callstack_item> newItem)
+            {
```

Comment:
The opening braces must be inline with the function declaration:
```suggestion
            auto add_callstack_item(ref<callstack_item> newItem) {
```